src/gui/presets.c

```diff
@@ -1575,7 +1575,8 @@ static void _gui_presets_popup_menu_show_internal(dt_dev_operation_t op,
             || (op_params_size > 0
                 && !memcmp(params, op_params, MIN(op_params_size, params_size))))
        && !memcmp(bl_params, blendop_params, MIN(bl_params_size, sizeof(dt_develop_blend_params_t)))
-       && module->enabled == enabled)
+       && ((module->enabled && enabled) || (!module->enabled && !enabled)) // safe check gboolean vs int32_t
```

Comment:
Not sure what this does? We set as active a non enabled preset (but a preset has always the enabled column set anyway) on a non enabled module.

Reply:
I was indeed thinking about this while changing...

Some lines above we have
```
const int32_t enabled = sqlite3_column_int(stmt, 6);
```

An here we test a gboolean vs an int23 read from database for being equal. Is that always correct?
The test is true if both are false/zero or both are true/non-zero.

Reply:
I think a int32_t is FALSE for 0 and TRUE for anything else.

Reply:
Yes but isn't `==` a test for identity? So in case `enabled` read from database is 2 in case of an error?

Reply:
Yes, I agree and this change is correct to me. What looks strange is the second part you've added:
`
|| (!module->enabled && !enabled)`

Reply:
Well the original test was also true if both were "false" because of `==`. Just the first part wouldn't be enough, no?

Reply:
I think it was a bug and that the meaning was : module->enabled == enabled == TRUE. I may be wrong,  but the other check doesn't really make sense to me.

Reply:
Probably you are right, that's probably what is intended: check for module is enabled **and** enabled in database

The original evaluated to: module is enabled **and logically same as** enabled in database.